AlphaWallet/Core/Types/RealmStore.swift

```diff
@@ -31,12 +32,18 @@ class RealmStore {
         thread.start()
     }
 
-    func performSync(_ callback: @escaping (Realm) -> Void) {
+    func performSync(_ block: @escaping (Realm) -> Void) {
         if Thread.isMainThread {
-            callback(mainThreadRealm)
+            block(mainThreadRealm)
         } else {
-            thread.performSync() {
-                callback(self.backgroundThreadRealm)
+            //NOTE: synchronize calls from different threads to avoid
+            //*** -[AlphaWallet.RunLoopThread performSelector:onThread:withObject:waitUntilDone:modes:]: target thread exited while waiting for the perform
+            dispatchPrecondition(condition: .notOnQueue(queueForRealmStore))
```

Comment:
This line crashes if it's not running on `queueForRealmStore`. Where do we enforce that `performSync(:)` (i.e. its caller) always runs on either `.main` or `queueForRealmStore `?

Reply:
the `else`  part could be called on any non main thread,  and we use 
```           
dispatchPrecondition(condition: .notOnQueue(queueForRealmStore))
queueForRealmStore.sync {
```
to make sure the no one other is trying to call in same time